client/router_create.go

```diff
@@ -78,6 +81,220 @@ func (cli *VanClient) getControllerRules() []rbacv1.PolicyRule {
 	return rules
 }
 
+func hashPrometheusPassword(password string) ([]byte, error) {
+	return bcrypt.GenerateFromPassword([]byte(password), 14)
+}
+
+func (cli *VanClient) GetVanPrometheusServerSpec(options types.SiteConfigSpec, van *types.RouterSpec, transport *appsv1.Deployment, siteId string) {
+	// prometheus-server container index
+	const (
+		prometheusServer = iota
+	)
+
+	van.PrometheusServer.Image = images.GetPrometheusServerImageDetails()
+	van.PrometheusServer.Replicas = 1
+	van.PrometheusServer.LabelSelector = map[string]string{
+		types.ComponentAnnotation: types.PrometheusComponentName,
+	}
+	van.PrometheusServer.Labels = map[string]string{
+		types.AppLabel:    types.PrometheusDeploymentName,
+		types.PartOfLabel: types.AppName,
+	}
+	for key, value := range van.PrometheusServer.LabelSelector {
+		van.PrometheusServer.Labels[key] = value
+	}
+	van.PrometheusServer.Annotations = options.Annotations
+	for key, value := range options.Labels {
+		van.PrometheusServer.Labels[key] = value
+	}
+
+	envVars := []corev1.EnvVar{}
+
+	sidecars := []*corev1.Container{}
+	volumes := []corev1.Volume{}
+	mounts := make([][]corev1.VolumeMount, 1)
+	err := configureDeployment(&van.PrometheusServer, &options.PrometheusServer.Tuning)
+	if err != nil {
+		fmt.Println("Error configuring prometheus server deployment:", err)
+	}
+	kube.AppendConfigVolume(&volumes, &mounts[prometheusServer], "prometheus-config", "prometheus-server-config", "/etc/prometheus")
+	kube.AppendSharedVolume(&volumes, []*[]corev1.VolumeMount{&mounts[prometheusServer]}, "prometheus-storage-volume", "/prometheus")
```

Comment:
The Prometheus server just needs an EmptyDirVolume, doesn't it? It is not going to share any volume with any other container, right?

Reply:
Yes, it is not sharing the volume used for the tsdb with any other container at this point.

Reply:
It would be better to have a new function that only creates an EmptyDirVolume, but definitely, this is not a blocker for this pull request.